### DISCUSSION TITLE
[Question]: What exactly does the preview quality parameter change?

### DISCUSSION THREAD

**Discussion by lukasig** [blakeblackshear/frigate#13345] [Ask A Question] [answered]
Parameter in question:

```
record:
  preview:
    quality: medium
```

The documentation says:

> Previews are low res / fps videos that are quickly scrubbable and can be used for notifications or time-lapses.

I have tried medium, high and very_high with absolutely no difference on the preview quality when scrubbing through timeline.

**Comment by hawkeye217** [comment]
Previews are generated every hour from a series of webp images. This setting sets the quality of those webp images. 

You may see an immediate change on previews for the current hour only. But making the change will not affect previews from the past.  

Note that these previews are always going to be very low resolution (this is by design), so making a change may not provide the kind of clarity a higher resolution video would.

**Reply by lukasig** [reply]
Does restarting frigate not "force" the generation of new previews to use the new config?

After every change in config I have restarted frigate with no visible change ton ew previews.

I will further test this and will follow with a message.

Thanks

**Reply by hawkeye217** [reply]
Previews are never regenerated.  So no, restarting will not force anything. 

If you change the setting mid hour, for example, you can expect the current hour’s preview to reflect the change immediately moving forward. Anything in the past from the moment of change will not be affected.

**Reply by lukasig** [reply]
Yes, I understood that. The previews that were already generated will not be updated if config changes.
I am saying that I updated the config from 'medium' to 'very_high', then did a frigate restart, and the previews generated **after** the restart are still the same quality as the other ones. Does it make sense?

**Reply by hawkeye217** [reply]
Yes, that makes sense. 

All this setting does is change the compression factor of the webp image. See the code: https://github.com/blakeblackshear/frigate/blob/453a8d794e31ccc428a73c6ac590d56af871f4d8/frigate/output/preview.py#L291

Depending on your scene, this may or may not visually affect the quality of the image.

**Reply by NickM-27** [reply, marked as answer]
To be clear it also affects the max bitrate allowed for the previews generation. But in any case if there is not much activity then the quality won't increase that much.

**Comment by lukasig** [comment]
As a follow-up question...

Is there any way of making sure previews are generated for the entire timeline as opposed from only when motion is detected?

my current config is:
```
record:
  preview:
    quality: medium
  retain:
    days: 0
    mode: all
  events:
    retain:
      default: 31
      mode: active_objects
```

When no motion is detected, no previews seem to be generated and a the message "No Preview Found" is displayed in the camera frame. 
I would like to have the option of scrubbing through the timeline with no interruption of the previews since they don't take much storage space anyway.

thanks

**Reply by hawkeye217** [reply]
Previews are retained according to your recording config’s retention policy. So you could do something like this:

```
record:
  preview:
    quality: medium
  retain:
    days: 3
    mode: motion
  events:
    retain:
      default: 31
      mode: active_objects

**Reply by lukasig** [reply]
In the config you suggested retain mode: motion. Since I want to have previews for the entire timeline, not only for when motion is detected, shouldn't the config be more like this?:
```
  retain:
    days: 3
    mode: all
```
but that would use way more storage space, even if i set it to "days: 1".

Can I not have "days: 0" and previews for "all"?

**Reply by NickM-27** [reply]
No you are misunderstanding, the mode doesn't affect previews. The reason no previews are saved is because the hour had no recordings saved. So by adding continuous recording in motion mode the previews will be saved

**Reply by hawkeye217** [reply]
I assumed you have a little motion on your cameras, but you are correct in that you will use more storage space and it will be cleaned up per your `days` setting. You can’t set it at 0 if you want to have previews for everything, as @NickM-27 said.

**Reply by lukasig** [reply]
I think that made it clear. It would be nice to have the option of tweaking that in a future version, but probably very few care about something like that anyway.

Thank you both for clarifying.

**Comment by darmach** [comment]
I don't want to clutter with another issue - @blakeblackshear may I bother you here a bit? Are what is shown when scrolling the history view? I set the quality to high already, but it's stil so so poor. Are they generated from detect or record stream - how could I increase the resolution? I would like to have more details when I scroll trough time looking for something.

**Reply by hawkeye217** [reply]
High-resolution scrubbing (grabbing the timeline handlebar and moving it up and down the timeline) would be bandwidth and resource intensive. To enable fast scrubbing, previews are low resolution by design. You can't change the resolution.

Once you release the handlebar and stop scrubbing, the high resolution recording begins playing at that point.

**Reply by NickM-27** [reply]
Yeah, to be clear with higher resolution it just wouldn't work at all, mobile phones are already at the edge of not being and to scrub fluently with the current quality

**Reply by darmach** [reply]
Ah, I was hoping I could tune something up - I have plenty enough bandwidth and fast raid storage underneath.
